src/lib/core/exception.cc

```diff
@@ -118,6 +136,34 @@ Exception::Exception(const struct type_info *type_arg, const char *file,
 		     exception_log, type_arg, file, line);
 }
 
+Exception *Exception::deep_copy() const
+{
+	Exception *head = NULL;
+	try {
+		head = dup();
+		head->cause = cause;
+
+		struct error *err = head;
+		struct error *cause;
```

Comment:
Please use a name different from member name.

Reply:
Ok. I changed the names.